autolens/analysis/aggregator/aggregator.py

```diff
@@ -229,6 +174,261 @@ def fit_interferometer_via_database_from(
     )
 
 
+class AbstractAgg:
+    def __init__(self, aggregator: af.Aggregator):
+        """
+        An abstract aggregator wrapper, which makes it straight forward to compute generators of objects from specific
+        samples of a non-linear search.
+
+        For example, in **PyAutoLens**, this makes it straight forward to create generators of `Tracer`'s drawn from
+        the PDF estimated by a non-linear for efficient error calculation of derived quantities.
+        Parameters
+        ----------
+        aggregator
+            An PyAutoFit aggregator containing the results of non-linear searches performed by PyAutoFit.
+        """
+        self.aggregator = aggregator
+
+    def make_object_for_gen(self, fit: Fit, galaxies: List[al.Galaxy]) -> object:

```

Comment:
Prefer @abstractmethod to raising NotImplemented - this results in an error being raised at instantiation rather than when this method is called if the method has not been implemented (fail early)